core/integration/container_test.go

```diff
@@ -227,6 +227,48 @@ func TestContainerExecWithWorkdir(t *testing.T) {
 	require.Equal(t, res.Container.From.WithWorkdir.Exec.Stdout.Contents, "/usr\n")
 }
 
+func TestContainerExecWithUser(t *testing.T) {
+	t.Parallel()
+
+	res := struct {
+		Container struct {
+			From struct {
+				User string
+
+				WithUser struct {
+					User string
+					Exec struct {
+						Stdout struct {
+							Contents string
+						}
+					}
+				}
+			}
+		}
+	}{}
+
+	err := testutil.Query(
+		`{
+			container {
+				from(address: "alpine:3.16.2") {
+					user
+					withUser(name: "daemon") {
```

Comment:
Can we add a test that this works with numeric uids too? It will still be a `string` type but like `"123"`.

I double checked and believe that is what buildkit supports underneath many layers, so worth checking it works for us up here too: https://github.com/sipsma/buildkit/blob/61307b8f2e73d681721b51856c10cb8606e5ea70/executor/oci/user.go#L20-L20

Reply:
Using numeric IDs works, but I'm not fully onboard with supporting it since relying on numeric IDs feels a little brittle, and the parameter is called `name`. Is there a use case for sticking with numeric IDs?

Reply:
Yeah use of a non-numeric `name` requires the presence of `/etc/passwd` (which buildkit reads to map the name to a UID), which not images have (minimal ones that just contain static binaries for instance). Also sometimes you don't want a dependency on `/etc/passwd` (which can change from base image to base image) and want to just have a consistent UID setting across many possible base images.